modules/references/server/controllers/reference.server.controller.js

```diff
@@ -378,43 +378,87 @@ exports.readMany = async function readMany(req, res, next) {
     const self = req.user;
 
     // build a query
-    const query = (function buildQuery() {
-      const query = {};
-
-      /**
-       * Allow non-public references only when userFrom or userTo is self
-       */
-      const isSelfUserFromOrUserTo =
-        self._id.equals(userFrom) || self._id.equals(userTo);
-      if (!isSelfUserFromOrUserTo) {
-        query.public = true;
-      }
-
-      /**
-       * Filter by userFrom
-       */
-      if (userFrom) {
-        query.userFrom = userFrom;
-      }
-
-      /**
-       * Filter by userTo
-       */
-      if (userTo) {
-        query.userTo = userTo;
-      }
-
-      return query;
-    })();
-
-    // find references by query
-    const references = await Reference.find(query)
-      .select(referenceFields)
-      .populate(
-        'userFrom userTo',
-        userProfile.userMiniProfileFields + ' created',
-      )
-      .exec();
+    const matchQuery = {};
+
+    // Allow non-public references only when userFrom or userTo is self
+    if (!self._id.equals(userFrom) && !self._id.equals(userTo)) {
+      matchQuery.public = true;
+    }
+
+    // Filter by userFrom
+    if (userFrom) {
+      matchQuery.userFrom = new mongoose.Types.ObjectId(userFrom);
+    }
+
+    // Filter by userTo
+    if (userTo) {
+      matchQuery.userTo = new mongoose.Types.ObjectId(userTo);
+    }
+
+    // Aggregate projection for User in reference
+    const userKeys = {
```

Comment:
This is basically `userProfile.userMiniProfileFields + ' created'`

I first started converting it to this format but then figured it might not be worth if:
- We'd soon use new Avatar API instead of returning most of this from the API
- We'd anyway add created + gender, thus making it custom.